rq/registry.py

```diff
@@ -226,15 +235,26 @@ def cleanup(self, timestamp: Optional[float] = None):
                     except NoSuchJobError:
                         continue
 
+                    if job.failure_callback:
+                        try:
+                            with self.death_penalty_class(CALLBACK_TIMEOUT, JobTimeoutException, job_id=job.id):
+                                job.failure_callback(job, self.connection,
+                                                     AbandonedJobError, AbandonedJobError(), traceback.extract_stack())
+                        except:  # noqa
+                            logger.exception('Registry %s: error while executing failure callback', self.key)
```

Comment:
Ideally I'd like to refactor this to `job.execute_failure_callback()` so we can have on method that handles execution of failure callbacks, currently it's also used here: https://github.com/rq/rq/pull/1824/files#diff-50984f276e8a6fea5f4ba4938afb6f4fbc544fd3751fbd16027dc0a326d6405eR1382 

Do you feel like refactoring this in this PR or leave it for later?

Reply:
Will do with pleasure, but a different PR will be better I think.
Started working on it - will open when done.

Reply:
@selwin see those 2 commits (I rebased on this branch so this is the diff)
https://github.com/rq/rq/pull/1851/files/2241493144c82727d5c56f7d56c7ff78642e0608..4bfae0bdc248632837d43166b145db519e6f81a4